src/main/java/com/breskul/bibernate/persistence/Session.java

```diff
@@ -124,19 +127,53 @@ public void close() {
   private void performDirtyChecking() {
     persistenceContext.getEntityKeys().stream()
         .filter(this::hasChanged)
+        .peek(entityKey -> log.trace("Found not flushed changes in the cache"))
         .forEach(this::flushChanges);
   }
 
   private <T> boolean hasChanged(EntityKey<T> entityKey) {
-    Object[] currentEntityState = EntityUtil.getEntityColumnValues(
-        persistenceContext.getEntity(entityKey));
+    Object[] currentEntityState = EntityUtil.getEntitySimpleColumnValues(
+        persistenceContext.getEntity(entityKey)).stream().map(Pair::right).toArray();
     Object[] initialEntityState = persistenceContext.getEntitySnapshot(entityKey);
-    return !Arrays.equals(currentEntityState, initialEntityState);
+
+    return !Arrays.equals(currentEntityState, initialEntityState) || isToOneRelationChanged(
```

Comment:
Just put  `|| isToOneRelationChanged(entityKey);` on the next line, will be prettier

Reply:
done